src/app/services/authentication/authentication.service.ts

```diff
@@ -26,17 +27,39 @@ export class AuthenticationService {
   }
 
   login(username: string, email: string, password: string) {
-    const headers = new HttpHeaders().set('Content-Type', 'application/json');
-    return this.http.post<any>(`${environment.sportsRestApiHost}/api/v1/login/`, { username, email, password }, { headers })
-    .pipe(map(user => {
-      localStorage.setItem(this.CURRENT_USER, JSON.stringify(user));
-      this.currentUserSubject.next(user);
-      return user;
-    }));
+    const headers = new HttpHeaders().set("Content-Type", "application/json");
+    return this.http
+      .post<any>(
+        `${environment.sportsRestApiHost}/api/v1/login/`,
+        { username, email, password },
+        { headers }
+      )
+      .pipe(
+        map(user => {
+          localStorage.setItem(this.CURRENT_USER, JSON.stringify(user));
+          this.currentUserSubject.next(user);
+          return user;
+        })
+      );
   }
 
   logout() {
+    const headers = new HttpHeaders();
+    headers.append("Content-Type", "application/json");
+    headers.append(
+      "Authorization",
+      JSON.parse(localStorage.getItem("currentUser")).key
+    );
     localStorage.removeItem(this.CURRENT_USER);
```

Comment:
Opcional: La líneas `localStorage.removeItem(this.CURRENT_USER);` y `this.currentUserSubject.next(null);` podrían llamarse dentro del "pipe" cuando la respuesta del end-point sea correcta. Esto solo es como para mantener un orden dentro del método

Reply:
Lo intenté, pero no sé por qué cuando los metí en el pipe se bugeo el logout, por lo cual lo dejé afuera. (bugs be like???)

Reply:
extraño, bueno, dejemoslo así